### PR TITLE
Added compression_mode field in google_compute_backend_bucket and google_compute_backend_service

### PR DESCRIPTION
**Pull request by AlfatahB**
Added compression_mode field in google_compute_backend_bucket and google_compute_backend_service resources.

fixes https://github.com/hashicorp/terraform-provider-google/issues/11251

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added support for `compression_mode` field in `google_compute_backend_bucket` and `google_compute_backend_service` resource
```

### CODE REVIEW

File: mmv1/products/compute/api.yaml [hunk 1]
```diff
@@ -727,6 +727,13 @@ objects:
                 name: 'headerName'
                 description: |
                   The header field name to match on when bypassing cache. Values are case-insensitive.
+      - !ruby/object:Api::Type::Enum
+          name: 'compressionMode'
+          description: |
+            The mode of compression to reduce the size of data sent from the Cloud CDN edge to the client.
```
Comment (rileykarson) [inline mmv1/products/compute/api.yaml:733]: Where did this description come from? I see "Compress text responses using Brotli or gzip compression, based on the client's Accept-Encoding header." in the API.

**Review by rileykarson — COMMENTED** [review]
Rather than introducing 2 samples tests per resource, can we use a single update test (https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/README.md#update-tests) per resource instead? Additionally, can we test removing the field from config once it has been set?

File: mmv1/products/compute/api.yaml (hunk 1 above)
Comment (AlfatahB) [reply mmv1/products/compute/api.yaml:733]: I didn't find the description in the [REST Reference Page](https://cloud.google.com/compute/docs/reference/rest/v1/backendBuckets) so I used the description of Dynamic Compression mentioned [here](https://cloud.google.com/cdn/docs/dynamic-compression#use-cases). Can you please share the link to the API doc you are referring to?

Comment (rileykarson) [reply mmv1/products/compute/api.yaml:733]: https://cloud.google.com/compute/docs/reference/rest/beta/backendBuckets

Comment (AlfatahB) [reply mmv1/products/compute/api.yaml]: Thanks, I've updated the description and added a single test as suggested.

**Review by rileykarson — APPROVED** [review]

**Comment by AlfatahB** [comment]
@rileykarson Could you please merge this PR?

**Pull request merged**